### ISSUE TITLE
[Bug]: Add numerical stability guard for SigLip2 logit_scale

### ISSUE THREAD

**Issue by prabhasg5** [kornia/kornia#3481]
### 🐛 Describe the bug

When calling SigLip2Model.forward() during training with an unconstrained logit_scale parameter that has grown to 100.0, the model produces NaN values in logits_per_image and logits_per_text instead of the expected similarity matrix with finite values.

### 🔄 Steps to Reproduce

```bash
import torch
from kornia.models.siglip2 import SigLip2Model, SigLip2Config

config = SigLip2Config()
model = SigLip2Model(config)

# 1.Simulate large logit_scale (can occur during training)
model.logit_scale.data.fill_(100.0)

pixel_values = torch.randn(2, 3, 224, 224)
input_ids = torch.randint(0, 1000, (2, 10))

# 2.Forward pass produces NaN without clamping
output = model(pixel_values=pixel_values, input_ids=input_ids)
print(output.logits_per_image)  # Contains NaN values
```

### 💻 Minimal Code Example

```python
import torch
from kornia.models.siglip2 import SigLip2Model, SigLip2Config

# Create model
config = SigLip2Config()
model = SigLip2Model(config)

# Simulate extreme logit_scale value (can occur during training)
model.logit_scale.data.fill_(100.0)  # Large positive value

# Prepare inputs
pixel_values = torch.randn(2, 3, 224, 224)
input_ids = torch.randint(0, 1000, (2, 10))

# Forward pass
output = model(pixel_values=pixel_values, input_ids=input_ids)

# Check for NaN (occurs without clamping)
print(f"logit_scale value: {model.logit_scale.item()}")
print(f"Effective scale (exp): {output.logit_scale.item()}")  # Should overflow
print(f"Contains NaN: {torch.isnan(output.logits_per_image).any()}")
print(f"logits_per_image:\n{output.logits_per_image}")
```

### ✅ Expected behavior

The model should maintain numerical stability regardless of learned parameter values.

### ❌ Actual behavior

exp(100) overflows, producing inf/NaN in logits and loss.

### 🔧 Environment

```shell
- Kornia version: 0.8.2
- PyTorch version: 2.9.1
- Python version: 3.11.14
- OS: macOS (darwin)
- Installation method: pip / from source
- CUDA/cuDNN version: N/A (CPU testing)
- GPU model: N/A
```

### 📝 Additional context

The overflow can occur during:
- Extended training runs where `logit_scale` learns extreme values
- Transfer learning scenarios with aggressive learning rates
- Any situation where the learnable temperature parameter grows beyond safe numerical bounds

### 🤝 Contribution Intent

- [x] I plan to submit a PR to fix this bug
- [ ] I'm reporting this bug but not planning to fix it

**Comment by RohitYandigeri** [comment]
Hi, thanks for the detailed report and repro!

I’d like to work on fixing this numerical stability issue by clamping the logit_scale before exponentiation (similar to CLIP-style implementations) and adding a regression test.

Please let me know if that approach sounds good.

**Comment by Iamabhipandat** [comment]
@egidioln hey can i work on this issue?

**Comment by jbbqqf** [comment]
Hi! Doing a triage pass on open bugs — this one looks like it can be closed as **already fixed**.

Evidence:
- The clamp landed in commit [f4ee06c](https://github.com/kornia/kornia/commit/f4ee06cef75eea6bb7a9f95a8041f634586b25b5) on 2026-01-12: `kornia/models/siglip2/model.py:223` now does `self.logit_scale.clamp(min=0.0, max=self.logit_scale_max_log).exp()` before the matmul. The cached `logit_scale_max_log` is set in `__init__` at line 124 from `config.logit_scale_max` (default 100.0, [config.py:102](https://github.com/kornia/kornia/blob/main/kornia/models/siglip2/config.py#L102)).
- The fix shipped via PR #3480 ("fix(siglip2): clamp logit_scale to avoid overflow"), merged 2026-01-12 13:46 UTC — same day this issue was filed, which is probably why it slipped through.
- The repro in this issue (`model.logit_scale.data.fill_(100.0)` then forward) now produces finite logits on `main`: clamping `100.0` against `log(100.0) ≈ 4.605` keeps `exp()` well below overflow.

Since you (the original reporter) also authored the merged PR, I suspect this issue was meant to be auto-closed by the PR but wasn't linked. Worth closing so the two newcomers above (cc @RohitYandigeri, @Iamabhipandat) don't spend cycles re-implementing it. If I missed an edge case the clamp doesn't cover, happy to dig further.

---
*Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the file lines, commit, and PR cross-reference were verified manually against `main` at HEAD.*

**Comment by ducha-aiki** [comment]
@jbbqqf is correct, it is fixed
